### PR TITLE
Update to use ITensorMPS and compatibility with ITensorTDVP

### PR DESCRIPTION
**Pull request by emstoudenmire**
Adds `using ITensorMPS` in key places such as examples. Update certain using statements to bring in non-exported methods or types from ITensorMPS. Remove `ITensorQTT.linsolve` since it has exactly the same signature as `ITensorTDVP.linsolve` (was causing a warning) and offers essentially the same functionality.

### CODE REVIEW

**Comment by mtfishman** [comment]
Seems like something funny happened with the formatting. We should move a formatting file (https://github.com/ITensor/ITensors.jl/blob/ffbb1f94c8f3f74a195a4cd8ccfce0932351142f/.JuliaFormatter.toml) into this repository.

File: Project.toml [hunk 1]
```diff
@@ -4,15 +4,17 @@ authors = ["Matthew Fishman <mfishman@flatironinstitute.org> and contributors"]
 version = "0.0.1"
 
 [deps]
-ITensorTDVP = "25707e16-a4db-4a07-99d9-4d67b7af0342"
+ITensorMPS = "0d1a4710-d33b-49a5-8f18-73bdf49b47e2"
 ITensors = "9136182c-28ba-11e9-034c-db9fb085ebd5"
+ITensorTDVP = "25707e16-a4db-4a07-99d9-4d67b7af0342"
```
Comment (mtfishman) [inline Project.toml:9]: Can we remove the `ITensorTDVP` dependency since it is superceded by `ITensorMPS`?

File: src/dft.jl [hunk 2]
```diff
@@ -1,123 +1,133 @@
+using ITensorMPS: apply, MPO
```
Comment (mtfishman) [inline src/dft.jl:1]:
```suggestion
using ITensorMPS: MPO, apply
```
It's conventional to list types first followed by functions.

**Comment by mtfishman** [comment]
I remember now that in the `linsolve` function defined in this repository I was experimenting with things like preconditioning to help with convergence in problems that were ill-conditioned. However, that doesn't justify have a different `linesolve` implementation here, and should be done in a more systematic way in the future in `ITensorMPS.jl` and/or `ITensorNetworks.jl`. So I agree with your assessment that this one should be removed.

**Comment by emstoudenmire** [comment]
Yes, I was wondering why when I ran formatting, so many files changed. The missing .toml file explains it.

**Comment by mtfishman** [comment]
Thanks, I guess CI is disabled right now which I'll have to investigate later.

Have you run some tests locally to see that things are still working? If so I'll go ahead and merge this.

**Comment by emstoudenmire** [comment]
I did just catch one issue (forgot to also remove `using ITensorTDVP` but did that now). And with that change, tests are passing on my machine.

**Comment by mtfishman** [comment]
Seems like the compat entries were pretty outdated as well, I updated them. I'll merge, thanks!

**Pull request merged**